cogito/protocol.go

```diff
@@ -212,18 +216,43 @@ func (src *Source) UnmarshalJSON(data []byte) error {
 // Validate verifies the Source configuration and applies defaults.
 func (src *Source) Validate() error {
 	//
-	// Validate mandatory fields.
+	// Evaluate mandatory fields.
 	//
 	var mandatory []string
-	if src.Owner == "" {
-		mandatory = append(mandatory, "owner")
+
+	defaultSinks := []string{"gchat", "github"}
+	defaultSinksSet := sets.From(defaultSinks...)
+	sinksSet := sets.From(src.Sinks...)
+
+	if sinksSet.Size() > 0 {
+		// First validate sinks are known and supported.
+		sinksNotValid := sinksSet.Difference(defaultSinksSet)
+		if sinksNotValid.Size() > 0 {
+			return fmt.Errorf("source: invalid sink: %s. Supported sinks: %s", sinksNotValid, defaultSinks)
+		}
 	}
-	if src.Repo == "" {
-		mandatory = append(mandatory, "repo")
+
+	if sinksSet.Size() == 0 || sinksSet.Contains("github") {
+		// No sinks implies backward compatibility mode where github is mandatory and gchat optional.
+		if src.Owner == "" {
+			mandatory = append(mandatory, "owner")
+		}
+		if src.Repo == "" {
+			mandatory = append(mandatory, "repo")
+		}
+		if src.AccessToken == "" {
+			mandatory = append(mandatory, "access_token")
+		}
 	}
-	if src.AccessToken == "" {
-		mandatory = append(mandatory, "access_token")
+
+	if sinksSet.Size() > 0 && sinksSet.Contains("gchat") {
```

Comment:
This is a breaking change compared to current behavior, no? If you want to use `github` and `gchat`, now you are required to set the sinks explicitly in source...

what happens if you don't set any sinks and you want use both `github` and `gchat`?

Or am I missing something here?

Reply:
Or I was wrong, if we don't set any sinks webhook is just optional param.

Reply:
It's 100% backward compatible, the default is applied, that means `github, gchat` are available. Note that this validation runs around the source configuration and not the put parameters. 

If you don't specify any sink, `repo,owner,access_token` are mandatory because Cogito default to its original role. In the other hand if you specify `gchat` you are requested to set the webhook.

Reply:
I understand your point. Its still backward compatible. This is my understanding:

- so if no sinks are configured we go to default (current behavior) and `gchat_webhook` is optional. If it is set you will post to Gchat if it isn't set it will not post to Gchat.

- but than if Gchat is configured the `gchat_webhook` is mandatory.

So in both cases we can post to Gchat.